src/trust_is_risk.js

```diff
@@ -0,0 +1,282 @@
+  async getTrustIncreasingMTX(origin : Key, dest : Key, outpoint : bcoin$Outpoint,
+      trustAmount : number, fee : ?number)
+      : Promise<bcoin$MTX> {
+    if (!fee) fee = 1000; // TODO: estimate this
+    var coin = await this.node.getCoin(outpoint.hash, outpoint.index);
+    if (!coin) throw new Error("Could not find coin");
+
+    var originKeyRing = KeyRing.fromPrivate(origin);
+    var originPubKey = originKeyRing.getPublicKey();
+
+    var mtx = new MTX({
+      outputs: [
+        new Output({
+          script: bcoin.script.fromMultisig(1, 2, [originPubKey, dest]),
+          value: trustAmount
+        })
+      ]
+    });
+
+    var changeAmount = coin.value - trustAmount - fee;
+    if (changeAmount) {
+      mtx.addOutput(new Output({
+        script: bcoin.script.fromPubkeyhash(bcoin.crypto.hash160(originPubKey)),
+        value: changeAmount
+      }));
+    }
+
+    mtx.addCoin(coin);
+    var success = mtx.scriptVector(coin.script, mtx.inputs[0].script, originKeyRing);
+    assert(success);
+
+    var signedCount = mtx.sign(originKeyRing);
+    assert(signedCount === 1);
+
+    return mtx;
+  }
+
+  // Returns an array of trust-decreasing mutable transaction objects, which reduce a trust
+  // relationship by the amount specified. The payee will receive the amount deducted minus the
+  // transaction fees via P2PKH.
+  // If steal is undefined or set to false, then the `origin` key is expected to be a private key
+  // and the `dest` key is expected to be a public key. If steal is set to true, then `origin` is
+  // expected to be a public key and `dest` is expected to be a private key. The private key will be
+  // used to sign the transaction.
+  getTrustDecreasingMTXs(origin : Key, dest : Key, trustDecreaseAmount : number, payee : ?Entity,
+      steal : ?boolean, fee : ?number) : bcoin$MTX[] {
+    if (steal === undefined) steal = false;
+
+    var signingKeyRing, originKeyRing, destKeyRing;
+    if (!steal) {
+      signingKeyRing = KeyRing.fromPrivate(origin);
+      originKeyRing = KeyRing.fromPrivate(origin);
+      destKeyRing = KeyRing.fromPublic(dest);
+    } else {
+      signingKeyRing = KeyRing.fromPrivate(dest);
+      originKeyRing = KeyRing.fromPublic(origin);
+      destKeyRing = KeyRing.fromPrivate(dest);
+    }
+
+    var originAddress = helpers.pubKeyToEntity(originKeyRing.getPublicKey());
+    var destAddress = helpers.pubKeyToEntity(destKeyRing.getPublicKey());
+
+    if (originAddress === destAddress) throw new Error("Can't decrease self-trust");
+
+    var existingTrustAmount = this.db.getDirectTrustAmount(originAddress, destAddress);
+    if (existingTrustAmount < trustDecreaseAmount) throw new Error("Insufficient trust");
+    
+    var directTrusts = this.db.getSpendableDirectTrusts(originAddress, destAddress);
+    return directTrusts.map((directTrust) => {
+      var decrease = Math.min(trustDecreaseAmount, directTrust.amount);
+      if (decrease === 0) return null;
```

Comment:
Perhaps a `for` loop with a `break` here would be more readable, also avoiding the `filter(Boolean)` part – and as a side effect it will also avoid unnecessary iterations.

Reply:
Apparently flow isn't that smart, can't get anything other than filter(Boolean) to work.